---
layout: post
title: "Binder design competition hosted by Adaptyv Bio"
date: 2025-04-07
mathjax: true
---

This note is a summary of a binder design competition hosted by Adaptyv Bio. Adaptyv is a company that bills itself as a wetlab partner to computational biology clients, i.e. it performs in-vitro testing of in-silico designs. Convenient for computational-only biotech firms. It is useful to analyze the format and metrics of this competition as well the results because it tells us what approaches are useful for a typical design problem. The competition took place in 2 stages, with the results from stage 1 informing work and submissions for stage 2. 

**Competition Goal**

The goal of this competition was to design novel single-chain proteins that bind the extracellular domain of EGFR (Epidermal Growth Factor Receptor), a critical therapeutic target in cancer. Designs had to be at least 10 amino acids different from any known therapeutic antibodies.

Entries (up to 10 per designer) were ranked using three computational metrics:

- AlphaFold2 iPAE interaction (interface Predicted Aligned Error) -- see glossary. Basically, AlphaFold2 produces a residue x residue matrix that measures how uncertain the model is about the location of residue j when the i^th residue is aligned to the true structure. iPAE specifically focused on (i,j) pairs across the binding interface, i.e. i belongs to the target and j belongs to the designed structure. 

- AlphaFold2 ipTM (inter-chain predicted TM-score): derived from AF2’s Predicted TM-score (pTM), which measures global structural confidence. ipTM specifically measures how confidently AF2 predicts the relative orientation and positioning between chains in a complex. The EGFR binder is one chain, EGFR is another. High ipTM in AF2 predicts a well-defined, consistent interface between binder and EGFR.

- ESM2 log-likelihood (how "natural" the sequence looks): This is simply the log likelihood of the designed protein sequence according to the ESM model. Since ESM is a language model on protein sequences, it is possible to assign a probability to any arbitrary protein sequence. Technically ESM's score is a pseudolikelihood because ESM is not an autoregressive language model and therefore the prob(sequence) cannot be computed as a product $$P(x1)p(x2|x1)p(x3|x1, x2)$$... and instead it must be computed as $$p(seq) = \Pi p(x_i | x_{-i}$$)

The average rank across these 3 metrics determined the top 100 selections for experimental validation, with an additional 100 chosen for novelty or creative methodology.

**Competition Outcomes & Statistics**

400 designs were tested in Adaptyv’s high-throughput wet lab: 100 from top computational rankings and 300 additional designs chosen for diversity. Results were quite interesting:

- 95% expression success rate: i.e. 95% of the proteins tested in the wetlab could actually be expressed in cells. 
- 14% binders (53 designs) showed detectable binding to EGFR in a biochemical assay. In academic design competitions, the binding hit rate can be pretty low - 1-2 %. 
- Some binders matched or exceeded Cetuximab’s performance (Merck's clinically approved antibody)


**Design approaches**

The most interesting part of the results is to analyze the approaches that participants took. Designing a binder for EGFR is not a toy problem -- it has infact been targetted by the South Korean AI company Galux using a _de novo_ approach, because EGFR is found exclusively on the surface of some cancer cells and not on other ambient proteins in the human body. 

The main design strategies can broadly be categorized into:

- De novo + fliter: De novo generation using say e.g., RFdiffusion backbone synthesis, possibly conditioned on the target geometry, followed by a _filter_ that would only keep some of the designs, and this filter typically consists of: (i) an inverse folding stage like ProteinMPNN that takes 3d structures generated by RFDiffusion and produces sequences, and (ii) using AlphaFold's predicted metrics to decide whether a given sequence + structure is 'good'. De novo design is kind of like a holy grail but it remain difficult because of a low _hit rate_ -- i.e. most designs do not actually bind or do not have other desirable properties. The filtering stage is therefore crucial - and even after that the remaining designs might not work _in vitro_. 

- Diversification of existing binders + filtering by metrics: start with known binders from literature, and mutate them, and then use AlphaFold metrics to decide whether the new protein is 'good' (stable, etc.). What does this actually mean? 

- Hallucination via AlphaFold2 optimization: Start from a random or partially defined sequence. Iteratively optimize the sequence so that AF2 predicts a folded structure with the desired binding interface. This “hallucinates” sequences that AF2 consistently interprets as structured binders.

- Unsupervised protein language model–guided optimization (e.g., active learning): PLMs score or propose mutations that are “natural-like” (likely to fold, stable). Combine with active learning: iteratively generate sequences, evaluate them (in silico or experimentally), and retrain the scoring function to bias towards better candidates.

- Physics-based docking and molecular dynamics, often followed by score filtering

I think an interesting problem in this space is how to systematize and unify these (and possibly other) approaches under an umbrella, so that it becomes clear what one should pursue for maximum / fast impact == being able to design good candidates with high hit rate, with high likelihood of success in the lab. 


Winner: Cradle — Optimization of Existing Binder. A variant of Cetuximab with 10 stabilizing framework mutations (not in the binding CDR loops). Achieved KD = 1.21 nM -- 8x better than a Cetuximab scFv (~9.9 nM), and 4x better than the nearest competitor (~5.2 nM). The cradle.bio group has a blog post about their winning entry here: https://www.cradle.bio/blog/adaptyv-protein-design-competition




