---
layout: page
title: Papers
includelink: true
---
A growing list of useful paper in the field. 

**Review papers**

- Protein therapeutics: a summary and pharmacological classification

- Scientific Discovery in the Age of Artificial Intelligence, Nature 2023

- [Hitchhiker’s guide to geometric GNNs for 3D atomic systems](https://arxiv.org/pdf/2312.07511.pdf)

- [Engineering protein-based therapeutics through structural and chemical design](https://www.nature.com/articles/s41467-023-38039-x)

- [Geometric deep learning book, Bronstein et al](https://arxiv.org/pdf/2104.13478.pdf)

- [Growing ecosystem of deep learning methods for modeling protein–protein interactions](https://arxiv.org/pdf/2310.06725.pdf)

- [Diffusion models in protein structure and docking](https://wires.onlinelibrary.wiley.com/doi/10.1002/wcms.1711)

- [The genetic Basis for Disease](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6279436/pdf/ebc-62-ebc20170053.pdf)

- [Genome-wide association studies](https://www.nature.com/articles/s43586-021-00056-9)



**Self-supervised learning on biological sequences**

- TAPE: Evaluating Protein Transfer Learning with TAPE, 2019

- [UCE: Universal Cell Embeddings](https://www.biorxiv.org/content/10.1101/2023.11.28.568918v1), Leskovec et al, 2023. 

- [scGPT: Towards Building a Foundation Model for Single-Cell 2 Multi-omics Using Generative AI](https://www.biorxiv.org/content/10.1101/2023.04.30.538439v1.full.pdf)

- [Learning a Continuous Representation of 3D Molecular Structures with Deep Generative Models](https://arxiv.org/pdf/2010.08687.pdf)

- [A Deep Dive into Single-Cell RNA Sequencing Foundation Models](https://www.biorxiv.org/content/10.1101/2023.10.19.563100v1.full.pdf)

- [scBERT as a large-scale pretrained deep language model for cell type annotation of single-cell RNA-seq data](https://www.nature.com/articles/s42256-022-00534-z.pdf)

- [ProtTrans: Towards Cracking the Language of Life’s Code Through Self-Supervised Learning](https://www.biorxiv.org/content/10.1101/2020.07.12.199554v3.full.pdf), 2021.

- [GeneFormer paper: Transfer learning enables predictions in network biology](https://www.nature.com/articles/s41586-023-06139-9)




**Protein Folding**
- AlphaFold2 paper: [Highly accurate protein structure prediction with AlphaFold](https://www.nature.com/articles/s41586-021-03819-2) J. Jumper et al, 2021

- Rosettafold paper: [Accurate prediction of protein structures and interactions using a three-track neural network](https://www.science.org/doi/10.1126/science.abj8754), Baek et al, 2021. 

- RosettaFold all atom paper: [Generalized biomolecular modeling and design with RoseTTAFold All-Atom](https://www.science.org/doi/10.1126/science.adl2528), Krishna et al, 2024. 

- [FoldingDiff: Protein structure generation via folding diffusion](https://www.nature.com/articles/s41467-024-45051-2?fromPaywallRec=false)

- ESM2 + ESMFold paper: [Evolutionary-scale prediction of atomic level protein structure with a language model](https://www.biorxiv.org/content/10.1101/2022.07.20.500902v2.full.pdf)

- [OmegaFold: High-resolution de novo structure prediction from primary sequence](https://www.biorxiv.org/content/10.1101/2022.07.21.500999v1), 2022





**Generative Models of molecules**
- [Robust deep learning–based protein sequence design using ProteinMPNN](https://www.biorxiv.org/content/10.1101/2022.06.03.494563v1.full.pdf), Dauparas et al.

- [RFdiffusion paper](https://www.biorxiv.org/content/10.1101/2022.12.09.519842v1)

- RFdiffusion all atom paper: [Generalized biomolecular modeling and design with RoseTTAFold All-Atom](https://www.science.org/doi/10.1126/science.adl2528), Krishna et al, 2024. 

- Chroma paper: [Illuminating protein space with a programmable generative model](https://www.nature.com/articles/s41586-023-06728-8), Ingraham et al, 2023. 

- [Generative Flows on Discrete State-Spaces: Enabling Multimodal Flows with Applications to Protein Co-Design](https://arxiv.org/pdf/2402.04997.pdf), 2024. 

- [Progen: Large language models generate functional protein sequences across diverse families](http://cdn.fraserlab.com/publications/2023_madani.pdf), Madani et al, 2023.

- [Diffusion models in protein structure and docking](https://wires.onlinelibrary.wiley.com/doi/10.1002/wcms.1711), Yim et al, 2024. 

- [Protpardelle - An all-atom protein generative mode](https://www.biorxiv.org/content/10.1101/2023.05.24.542194v1.full.pdf)

- [Equivariant Diffusion for Molecule Generation in 3D](https://arxiv.org/abs/2203.17003), for small moledules, equivariant to Euclidean transformations. 

- [Generative Flows on Discrete State-Spaces: Enabling Multimodal Flows with Applications to Protein Co-Design](https://arxiv.org/abs/2402.04997)

- FrameDiff paper: [SE(3) diffusion model with application to protein backbone generation](https://arxiv.org/pdf/2302.02277), Jason Yim et al. 

- SCMDiff paper: [Diffusion probabilistic modeling of protein backbones in 3D for the motif-scaffolding problem](https://arxiv.org/abs/2206.04119), Trippe et al, 2022. 






**Miscellaneous**
- [Learning the Drug-Target Interaction Lexicon](https://www.biorxiv.org/content/10.1101/2022.12.06.519374v1.full.pdf)

- [Protein Design with Guided Discrete Diffusion](https://arxiv.org/pdf/2305.20009.pdf)


**Geometric Deep learning**
- [SE(3)-Transformers: 3D Roto-Translation Equivariant Attention Networks](https://arxiv.org/pdf/2006.10503)

- [Hitchhiker’s guide to geometric GNNs for 3D atomic systems](https://arxiv.org/pdf/2312.07511.pdf)

- [Geometric deep learning book, Bronstein et al](https://arxiv.org/pdf/2104.13478.pdf)

- Framediff paper: [SE(3) diffusion model with application to protein backbone generation](https://arxiv.org/pdf/2302.02277), Jason Yim et al.


